File: libs/user-authentication/verification-code/src/main/kotlin/com/alesharik/user/authentication/verification/code/exception/VerificationCodeNotRequestedException.java
package com.alesharik.user.authentication.verification.code.exception;

import org.zalando.problem.Status;
import org.zalando.problem.StatusType;
import org.zalando.problem.ThrowableProblem;

import java.net.URI;

public class VerificationCodeNotRequestedException extends ThrowableProblem {
    public static final URI TYPE = URI.create("https://spring.alesharik.com/docs/error/auth/verification-code-not-requested");

    @Override
    public URI getType() {
        return TYPE;
    }

    @Override
    public String getTitle() {
        return "Verification code not requested";
    }

    @Override
    public String getDetail() {
        return "You should request verification code from your auth handler";
    }

    @Override
    public StatusType getStatus() {
        return Status.BAD_REQUEST;
    }
}

File: libs/user-authentication/verification-code/src/main/kotlin/com/alesharik/user/authentication/verification/code/exception/VerificationCodeNotRequestedException.kt
package com.alesharik.user.authentication.verification.code.exception

import org.springframework.http.HttpStatus
import org.springframework.web.ErrorResponseException
import java.net.URI

class VerificationCodeNotRequestedException : ErrorResponseException(HttpStatus.BAD_REQUEST) {
    init {
        setType(TYPE)
        setTitle("Verification code not requested")
        setDetail("You should request verification code from your auth handler")
    }

    companion object {
        val TYPE = URI.create("https://spring.alesharik.com/docs/error/auth/verification-code-not-requested")
    }
}
